pages/en/android/camera-orientation.md

```diff
@@ -0,0 +1,301 @@
+| 270              | ![A portrait-shaped image with the bugdroid's head pointed to the left](/images/android/camera-orientation/texture_view_display_rotation_270_phone.png) | ![A landscape-shaped image with the bugdroid's head pointed to the left](/images/android/camera-orientation/texture_view_display_rotation_270_laptop.png) |
+
+#### Layout
+
+Layout is a bit tricky for the case of TextureView. It has been previously suggested to use a transformation matrix for TextureView, but that method does not work for all devices. We suggest that you follow the steps described here instead.
+
+The 3-step process to correctly layout previews on a TextureView:
+
+1. Set the size of the TextureView to be identical to the preview size chosen.
+2. Scale the potentially stretched TextureView back to the original dimensions of the preview.
+3. Rotate the TextureView by `displayRotation` counterclockwise.
+
+Suppose you have a phone with a display rotation of 90 degrees.
+
+![Illustration of a phone with a display rotation of 90 degrees, and an object](/images/android/camera-orientation/texture_view_observer_side.png)
+
+**1. Set the size of the TextureView to be identical to the preview size chosen**
+
+Suppose the preview size you chose is `previewWidth × previewHeight` where `previewWidth > previewHeight` (sensor output is landscape-shaped by nature). When configuring a capture session, one should call [SurfaceTexture#setDefaultBufferSize(int width, height)](<https://developer.android.com/reference/android/graphics/SurfaceTexture#setDefaultBufferSize(int,%20int)>) to specify the preview size (`previewWidth × previewHeight`).
+
+Before calling setDefaultBufferSize, it’s important that you **also set the size of the TextureView to be `previewWidth × previewHeight`** with [View#setLayoutParams(android.view.ViewGroup.LayoutParams)](<https://developer.android.com/reference/android/view/View#setLayoutParams(android.view.ViewGroup.LayoutParams)>). The reason for this is that TextureView calls `SurfaceTexture#setDefaultBufferSize(int width, height)` with its measured width and height. If the size of the TextureView is not explicitly set beforehand, it can cause a race condition. This is mitigated by explicitly setting the size of the TextureView first.
+
+Now the TextureView may not match the dimensions of the source. In the case of phones, the source is portrait-shaped, yet the TextureView is landscape-shaped owing to the layoutParams you just set. This would result in a stretched previews, as illustrated below,
+
+![Image illustration of a portrait-shaped preview stretched to fit inside a TextureView of the same size of the preview size chosen](/images/android/camera-orientation/texture_view_step_1_identical_size.png)
+
+##### 2. Scale the potentially stretched TextureView back to the original dimensions of the preview
```

Comment:
Please ensure these items in the process are consistently headers or just bold text in a paragraph, and please don't include numbers if they're headers.

Reply:
Here we're trying to show a 3-step process, so it's procedural.

The problem here is each of the 3 steps needs quite a bit of paragraph, and since it's procedural, I think it makes sense to include numbers to indicate which step we're in.

Let me think how this can be dealt with within the constraints of Markdown.

Reply:
Corrected :) Thanks!